llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -758,18 +758,16 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
           all(typeInSet(0, {s8, s16, s32, s64, s128}), typeIs(2, p0)));
 
   getActionDefinitionsBuilder(G_ATOMIC_CMPXCHG)
-      .customIf([](const LegalityQuery &Query) {
-        return Query.Types[0].getSizeInBits() == 128;
-      })
-      .clampScalar(0, s32, s64)
-      .legalIf(all(typeInSet(0, {s32, s64}), typeIs(1, p0)));
+      .legalFor({{s32, p0}, {s64, p0}})
+      .customIf(typeIs(0, s128))
```

Comment:
Looks like rule on 128-bit vector types case was removed, but not sure if it was needed. Is the patch meant to be NFC?

Reply:
(It might be better to keep it in just to be explicit about the types that are supported.)

Reply:
Sorry I think I was looking at an older version. I think you could probably use customFor({s128, p0}), but either way this looks OK to me.

Reply:
@ilinpv pointed out that testing for s128 is not equivalent to what was done so I've changed to keep the old test to keep this more of an RFC. Strangely, it required updating the legalizer-info-validation test. Would you happen to know why that changed?